AgiliTEAM/pacman_env.py

```diff
@@ -0,0 +1,282 @@
+from typing import Tuple
+import cv2
+import numpy as np
+np.random.seed(0)
+
+MAP_SIZE = 30
+MAX_STEP = 100
+NUM_GHOSTS = 4
+NUM_COINS = 100
+ACTION_SPACE_SIZE = 4
+SCORE_COIN = 10
+DIRECTIONS = {"up": 0, "right": 1, "down": 2, "left": 3}
+
+
+class Pacman:
+    """
+    ------------------------
+    Pacman Environment Class
+    ------------------------
+
+
+    Attributes:
+        map_size : int
+            size of the square map's one side
+        action_space : ndarray
+            defines the executable actions in the environment
+        coins_pos : list (of tuples)
+            contains the (x, y) coordinate pairs of the generated coin objects
+        ghosts_pos : list (of tuples)
+            contains the (x, y) coordinate pairs of the generated ghost objects
+        pos : tuple
+            defines the position of PacMan
+        step_counter : int
+            keeps track of the executed steps throughout the game
+        score : int
+            keeps track of the points acquired throughout the game
+        orientation : int
+            defines PacMan's facing direction
+        state : ndarray
+            defines an array representing the game status
+
+
+    Methods:
+        __init__(self, image_size: int = 600) -> None
+        reset(self) -> numpy.ndarray
+        step(self, action: int) -> Tuple[numpy.ndarray, float, bool, str]
+        render(self) -> None
+        _move(self) -> None
+        _get_info(self) -> str
+        _get_reward(self, state: numpy.ndarray) -> float
+        _get_next_state(self, action: int) -> numpy.ndarray
+        _set_action(self, action: int | None) -> None
+        _check_action_validity(self, action: int) -> None
+        _calculate_score(self) -> None
+        _check_done(self) -> bool
+        _generate_pos(self, object_name: str, num_objects: int) -> None
+        _update_map(self) -> numpy.ndarray
+    """
+    def __init__(self, image_size: int = 600) -> None:
+        """
+        Constructs the basic components of the Pacman Environment Class.
+        :return: None
+        """
+        self.map_size = MAP_SIZE
+        self.action_space = np.arange(ACTION_SPACE_SIZE)
+
+        self.coins_pos = []
+        self.ghosts_pos = []
+        self.pos = None
+
+        self.step_counter = None
+        self.score = None
+        self.orientation = None
+
+        self.image_size = image_size
+        self.image = np.zeros((self.image_size, self.image_size, 3))
+        self.ratio = int(self.image_size / self.map_size)
```

Comment:
Image_size, image and ratio are class variable not shown in the docstring of the class.